docs/poc2023q1/outline.md

```diff
@@ -0,0 +1,112 @@
+- Propagation of reported state from edge to center.
+- Summarization of reported state in the center.
+
+Some important things that are not attempted in this PoC include the following.
+
+- An implementation that supports a large number of edge clusters.
+- More than one SyncTarget per Location.
+- A hierarchy with more than two levels.
+
+It is TBD whether the implementation will support intermittent
+connectivity.  This depends on whether we can quickly and easily get a
+syncer that creates the appropriately independent targets and itself
+tolerates intermittent connectivity.
+
+Also: initially this PoC will not transport non-namespaced objects.
+CustomResourceDefinitions are thus among the objects that will not be
+transported.  Transport of non-namespaced objects may be added after
+the other goals are achieved.
+
+## Design overview
+
+In very brief: the design is to reduce each edge placement problem to
+many instances of kcp's TMC problem.
+
+See [the overview picture](Edge-PoC-2023q1.svg) for an overview
+picture.
+
+## Inventory Management workspaces
+
+This design takes as a given that something maintains some kcp
+workspaces that contain dynamic collections of Location and SyncTarget
+objects as defined in [kcp
+TMC](https://github.com/kcp-dev/kcp/tree/main/pkg/apis), and that one
+view can be used to read those Location objects and one view can be
+used to read those SyncTarget objects.
+
+## Edge Service Provider workspace
+
+The edge multi-cluster service is provided by one workspace that
+includes the following things.
+
+- An APIExport of the edge API group.
+- The edge controllers: scheduler, placement translator, mailbox
+  controller, and status sumarizer.
+
+## Workload Management workspaces
+
+The users of edge multi-cluster primarily maintain these.  Each one
+contains the following things.
+
+- A workload spec.  That is, some namespaces containing namespaced
+  workload objects --- and possibly later on some non-namespaced
+  workload objects.  Initially the workload objects must be of types
+  drawn from a list of those supported by the EMC implementation
+  (because extension by user CRDs is not supported yet).
+- EMC interface objects: EdgePlacement, SinglePlacementSlice, those
+  for customization and summarization.
+
+## Mailbox Controller
+
+This controller maintains one mailbox workspace per SyncTarget.  Each
+of these mailbox workspaces is used for a distinct TMC problem (e.g.,
+Placement object).  These workspaces are all children of the edge
+service provider workspace.
+
+## Edge Scheduler
+
+This controller monitors the EdgePlacement, Location, and SyncTarget
+objects and maintains the results of matching.  For each EdgePlacement
+object this controller maintains an associated collection of
+SinglePlacementSlice objects holding the matches for that
+EdgePlacement.  These SinglePlacementSlice objects appear in the same
+namespace as the corresponding EdgePlacement; the remainder of how
```

Comment:
Should it be "workspace" instead of "namespace" ? Are EdgePlacement and SinglePlacementSlice non-namespaced objects ?

Reply:
Yes, it was a typo.  Thanks.  Fixed in the latest rev.